### PR TITLE
Removes references to role_special

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Removes the confusing and poorly named role_special "special additional antagonists" role preference, since if someone is using this to add a new antag, they should make their own role_[antag] preference for it.

## Why It's Good For The Game

Confusion UI that shouldn't have been used anyway, since you should define role_[antag] anyway yourself.

## Changelog
:cl:
code: removed confusing and poorly named preference type.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Edit: Not sure how this will affect people already banned from the role.

File: code/__DEFINES/role_preferences.dm [hunk 1]
```diff
@@ -37,7 +37,6 @@
 #define ROLE_LAVALAND			"Lavaland"
 #define ROLE_INTERNAL_AFFAIRS	"Internal Affairs Agent"
 #define ROLE_GANG				"Gangster"
-#define ROLE_SPECIAL			"Special Antagonists"
```
Comment (zeskorion) [inline code/__DEFINES/role_preferences.dm:40]: I don't see why you need to remove the defines and bits that use them themselves. Mebbe just change the names and remove the preference bits

File: code/modules/events/special_antag_event.dm [hunk 2]
```diff
@@ -34,7 +34,7 @@
 	fakeable = FALSE
 	var/role_name
 	var/antag_datum	//The datum of the antag E.G. /datum/antagonist/special/undercover
-	var/preference_type = ROLE_SPECIAL
+	var/preference_type = ROLE_TRAITOR
 	var/protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Head of Personnel", "Chief Medical Officer", "Chief Engineer", "Research Director", "Captain", "Brig Physician")
 
 /datum/round_event/create_special_antag/start()
```
Comment (zeskorion) [inline code/modules/events/special_antag_event.dm:41]: I've heard reports from admins about this event. Can it occur right now?

**Review by zeskorion — CHANGES_REQUESTED** [review]

File: code/modules/events/special_antag_event.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/events/special_antag_event.dm:41]: The fix just got merged

File: code/__DEFINES/role_preferences.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/__DEFINES/role_preferences.dm:40]: This is never used in the code, and antags made with this system should use their own defines

**Comment by ike709** [comment]
Conflicts

**Pull request merged**